### PR TITLE
Avoid auto-merging thirdparty/cudf-pins

### LINKED ISSUE #1927: [BUG] auto-merge between branches should not merge pinned cudf dependency versions
When automatically merging PRs between branches, there's logic in .github/workflows/auto-merge.yml to avoid merging changes to thirdparty/cudf.  We need to extend this to include avoiding merging changes to thirdparty/cudf-pins/.

### PR DESCRIPTION
**Pull request by jlowe**
Fixes #1927.  This avoids automatically merging thirdparty/cudf-pins files when automatically merging changes between branches.

### CODE REVIEW

**Comment by jlowe** [comment]
@NvTimLiu do you have a way to test this change?  I'm unclear on how the workflows get tested.

File: .github/workflows/auto-merge.yml [hunk 1]
```diff
@@ -42,13 +42,13 @@ jobs:
           git config user.email "70000568+nvauto@users.noreply.github.com "
           git fetch origin ${HEAD} ${BASE}
           git checkout -b ${INTERMEDIATE_HEAD} origin/${HEAD}
-          OUT=$(git --no-pager diff --name-only origin/${BASE} | grep "${FILE_USE_BASE}" || true)
-          [[ ! -z "${OUT}" ]] && git checkout origin/${BASE} -- ${FILE_USE_BASE} && \
-            git commit -s -am "Auto-merge use submodule in BASE ref"
+          git checkout origin/${BASE} -- ${FILES_USE_BASE}
+          [ -z "$(git status --porcelain=v1 --untracked=no) ] && \
```
Comment (NvTimLiu) [inline .github/workflows/auto-merge.yml:46]: should be "$(git status --porcelain=v1 --untracked=no)"  ?

Comment (NvTimLiu) [reply .github/workflows/auto-merge.yml:46]: Auto merge code change : https://github.com/NvTimLiu/spark-rapids-jni/pull/73

Not merge submodule change: https://github.com/NvTimLiu/spark-rapids-jni/pull/75 (nothing changed on branch-24.06)

**Comment by NvTimLiu** [comment]
> @NvTimLiu do you have a way to test this change? I'm unclear on how the workflows get tested.

 This change have not yet be merged, and can not be triggered in the NVIDIA repo, so this auto-merge workflow change have to be verified in the forked repo instead

**Comment by NvTimLiu** [comment]
@jlowe Oh I intended to suggest change on your PR, but did not know how to add suggestions, so changed auto-merge.yml file on your branched, please feel free to discard/revert it if you've better fix. Thanks!

**Comment by YanxuanLiu** [comment]
build

**Comment by YanxuanLiu** [comment]
> build

@NvTimLiu Please help to trigger pre-merge pipeline. I still got not-authorized error. Thanks!

**Comment by jlowe** [comment]
build

File: .github/workflows/auto-merge.yml (hunk 1 above)
Comment (jlowe) [reply .github/workflows/auto-merge.yml]: Thanks!

File: .github/workflows/auto-merge.yml [hunk 2]
```diff
@@ -42,13 +42,13 @@ jobs:
           git config user.email "70000568+nvauto@users.noreply.github.com "
           git fetch origin ${HEAD} ${BASE}
           git checkout -b ${INTERMEDIATE_HEAD} origin/${HEAD}
-          OUT=$(git --no-pager diff --name-only origin/${BASE} | grep "${FILE_USE_BASE}" || true)
-          [[ ! -z "${OUT}" ]] && git checkout origin/${BASE} -- ${FILE_USE_BASE} && \
-            git commit -s -am "Auto-merge use submodule in BASE ref"
+          git checkout origin/${BASE} -- ${FILE_USE_BASE}
```
Comment (gerashegalov) [inline .github/workflows/auto-merge.yml:45]: Please add somewhere comments or a doc explaining at the high level what is happening here.

Comment (NvTimLiu) [reply .github/workflows/auto-merge.yml:46]: > Please add somewhere comments or a doc explaining at the high level what is happening here. Among others, it should reduce the chance of the previous confusion if we want an empty git status or not.

Added comments to explain what we want to do here

https://github.com/NVIDIA/spark-rapids-jni/pull/1928/commits/22314bc90be2315b58565058ef0b862b0a369e58

**Review by NvTimLiu — APPROVED** [review]

File: .github/workflows/auto-merge.yml [hunk 3]
```diff
@@ -42,13 +42,15 @@ jobs:
           git config user.email "70000568+nvauto@users.noreply.github.com "
           git fetch origin ${HEAD} ${BASE}
           git checkout -b ${INTERMEDIATE_HEAD} origin/${HEAD}
-          OUT=$(git --no-pager diff --name-only origin/${BASE} | grep "${FILE_USE_BASE}" || true)
-          [[ ! -z "${OUT}" ]] && git checkout origin/${BASE} -- ${FILE_USE_BASE} && \
-            git commit -s -am "Auto-merge use submodule in BASE ref"
+          # Use commits from HEAD, but keek submodule files(FILE_USE_BASE) from BASE
```
Comment (gerashegalov) [inline .github/workflows/auto-merge.yml:45]: nit: HEAD is confusing given that it also used for the tip of the branch
```suggestion
          # Sync the $BASE branch using commits from the $HEAD branch except for the paths  
          # defined as $FILE_USE_BASE (under ./thirdparty) 
```

**Review by gerashegalov — APPROVED** [review]
LGTM, non-blocking nit

File: .github/workflows/auto-merge.yml (hunk 3 above)
Comment (NvTimLiu) [reply .github/workflows/auto-merge.yml:45]: Sorry, let me follow up the typo in the comments here

**Pull request merged**